0-module-and-module-source.emu

```diff
@@ -83,6 +83,38 @@ location: https://tc39.es/proposal-compartments/
     </emu-table>
 </emu-clause>
 
+<emu-clause id="sec-meta-properties">
+  <h1>Meta Properties</h1>
+
+  <emu-clause id="sec-meta-properties-runtime-semantics-evaluation" type="sdo">
+    <h1>Runtime Semantics: Evaluation</h1>
+
+    <emu-grammar>ImportMeta : `import` `.` `meta`</emu-grammar>
+    <emu-alg>
+      1. Let _module_ be GetActiveScriptOrModule().
+      1. Assert: _module_ is a Source Text Module Record.
+      1. Let _importMeta_ be _module_.[[ImportMeta]].
+      1. If _importMeta_ is ~empty~, then
+        1. Set _importMeta_ to OrdinaryObjectCreate(*null*).
+        1. <ins>If _module_'s [[ModuleInstance]] internal slot value is *undefined*, then</ins>
+          1. Let _importMetaValues_ be HostGetImportMetaProperties(_module_).
+          1. For each Record { [[Key]], [[Value]] } _p_ of _importMetaValues_, do
+            1. Perform ! CreateDataPropertyOrThrow(_importMeta_, _p_.[[Key]], _p_.[[Value]]).
+          1. Perform HostFinalizeImportMeta(_importMeta_, _module_).
+          1. Set _module_.[[ImportMeta]] to _importMeta_.
+          1. Return _importMeta_.
+        1. <ins>Else,</ins>
+          1. <ins>Let _importMetaHook_ be _module_.[[ModuleInstance]].[[ImportMetaHook]].</ins>
```

Comment:
what if it is undefined?

Reply:
It can't be undefined.... if you have an instance, you will have a hook for import.meta.

Reply:
we can add an assertion here, but in the definition of module instance internal slots, it is clearly defined as a callable.

Reply:
concerned was addressed, it is now possible that this value is undefined, and we account for that.